Add sleep to OTEL exporters

jamesbaskerville opened the pull request:
Infinite polling the export queue without a sleep was causing threads to take up a full core of CPU. Simple sleep fixes that.

peadaroh commented on src/humanloop/otel/exporter.py:
```diff
@@ -153,6 +154,7 @@ def _do_work(self):
                 )  # type: ignore
             except EmptyQueue:
                 # Wait for the another span to arrive
```
fix typo

peadaroh commented on src/humanloop/otel/exporter.py:
```diff
@@ -153,6 +154,7 @@ def _do_work(self):
                 )  # type: ignore
             except EmptyQueue:
                 # Wait for the another span to arrive
+                time.sleep(0.1)
```
worth making it slightly longer? don't totally understand in what context this code is being used

jamesbaskerville replied:
Yeah it's fine to make it longer, too. There's a task in the queue whenever a span should be exported which approximately means every generation/evaluation/call. In an evaluations.run setting with up to 20 workers that could happen quite a few times a second, but would probably be spiky given they're usually dependent on provider calls.

(exporter has 4 workers by default I think)

harry-humanloop replied:
claude seems to give a range of 100-500ms as suitable generally when there's no need for realtime 
	•	Real-time systems: 10-50ms
	•	Interactive applications: 50-100ms
	•	Background processing: 100-500ms
	•	Batch processing: 500ms+

jamesbaskerville replied:
mmk, leaving as is sounds reasonable in that case

andreibratu commented:
Can you merge in the release branch?

jamesbaskerville committed:
fix: Add sleep to OTEL exporters

Infinite polling the export queue without a sleep was causing threads to take up a full core of CPU. Simple sleep fixes that.
